translator/common/otlp_to_oc.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	occommon "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	ocresource "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	otlpcommon "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1"
+	otlpresource "github.com/open-telemetry/opentelemetry-proto/gen/go/resource/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector/translator/conventions"
+)
+
+var (
+	defaultTime      = time.Time{} // zero
+	defaultTimestamp = timestamp.Timestamp{}
+	defaultProcessID = 0
+)
+
+func ResourceToOC(resource *otlpresource.Resource) (*occommon.Node, *ocresource.Resource) {
```

Comment:
Extracted from https://github.com/open-telemetry/opentelemetry-collector/blob/3dbff97772d186fa8f5155742735fc156ecf6a4b/translator/trace/otlp_to_oc.go#L173

Will migrate trace translator to using this function in a separate PR.

Reply:
I assume nothing changed in this func, it was just moved to a different file, right?

Reply:
Correct, no change.